### ISSUE TITLE
[libc] UB in llvm-project/libc/src/string/memory_utils/utils.h:assume_aligned

### ISSUE THREAD

**Issue by petrhosek** [llvm/llvm-project#64758]
UBSan running on RISC-V (Fuchsia) is reporting an undefined behavior at https://github.com/llvm/llvm-project/blob/e6cd409fc6396cb13c59b4a5190abc4b856f22a5/libc/src/string/memory_utils/utils.h#L87

```
assumption of 2 byte alignment for pointer of type 'const __llvm_libc::cpp::byte *' failed, address is 1 aligned
```

**Comment by gchatelet** [comment]
I somewhat missed this issue. @petrhosek  do you have more context about the failure or a way I can reproduce?

**Comment by petrhosek** [comment]
We believe the issue is this function: https://github.com/llvm/llvm-project/blob/a0029a8a765f2bfe938de2c34ffb0264724ca04a/libc/src/string/memory_utils/generic/aligned_access.h#L35-L46

This doesn't handle the case when alignment is 6. The `else` case handles all odd numbered alignments, and 6 just happens to fall into it. The following fixes the issue:

```cpp
 else if (alignment == 6)
    return load64_aligned<uint16_t, uint32_t, uint16_t>(ptr, offset);
```

This doesn't happen on other architectures as they don't use these generic functions for `memcpy`.

**Comment by gchatelet** [comment]
Thx a lot for pinpointing this down, I've created #71391 to fix it.
